qa/L0_server_status/server_status_test.py

```diff
@@ -111,17 +117,42 @@ def test_model_latest_infer(self):
 
             # Initially there should be no version stats..
             try:
-                for pair in [("localhost:8000", ProtocolType.HTTP), ("localhost:8001", ProtocolType.GRPC)]:
-                    server_status0, req_id0 = _get_server_status(pair[0], pair[1], model_name)
-                    self.assertTrue(model_name in server_status0.model_status,
-                                    "expected status for model " + model_name)
-                    self.assertEqual(len(server_status0.model_status[model_name].version_status), 3,
-                                    "expected status for 3 versions for model " + model_name)
+                for pair in [("localhost:8000", "http"), ("localhost:8001", "grpc")]:
+                    if pair[1] == "http":
+                        triton_client = httpclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+                    else:
+                        triton_client = grpcclient.InferenceServerClient(
+                            url=pair[0], verbose=True)
+
+                    self.assertTrue(triton_client.is_server_live())
+                    self.assertTrue(triton_client.is_server_ready())
+                    model_metadata = triton_client.get_model_metadata(model_name)
+                    # verify all versions are reported when no model version is specified
+                    if pair[1] == "http":
+                        self.assertEqual(model_name, model_metadata['name'])
+                        self.assertEqual(len(model_metadata['versions']), 3)
+                        for v in (1, 2, 3):
+                            self.assertTrue(str(v) in model_metadata['versions'])
+                    else:
+                        self.assertEqual(model_name, model_metadata.name)
+                        self.assertEqual(len(model_metadata.versions), 3)
+                        for v in (1, 2, 3):
+                            self.assertTrue(str(v) in model_metadata.versions)
+
                     for v in (1, 2, 3):
-                        self.assertTrue(v in server_status0.model_status[model_name].version_status,
-                                        "expected version " + str(v) + " status for model " + model_name)
-                        self.assertEqual(server_status0.model_status[model_name].version_status[v].ready_state,
-                                        server_status.MODEL_READY)
+                        self.assertTrue(triton_client.is_model_ready(
+                            model_name, model_version=str(v)))
+                        model_metadata = triton_client.get_model_metadata(
```

Comment:
Based on comments above this isn't going to work. We likely don't need so much testing around model metadata since it really is fairly simple. Really just need to make sure it returns the expected versions, input and output information. And that if we do call it with an unknown version that is fails.

Reply:
Add tests for model metadata contents and unknown model version. We might need a fix to the model_repository_manager to have a more meaningful error message when sending an incorrect version. I reports the error message as `"Request for unknown model <model_name>"` in the case the `model_version` is unknown even if the `model_name` is valid.

Reply:
OK, please file a 20.06 ticket for that.

Reply:
Done